typescript/ws-protocol/src/FoxgloveServer.ts

```diff
@@ -236,11 +264,47 @@ export default class FoxgloveServer {
 
         break;
 
+      case "advertise":
+        for (const channel of message.channels) {
+          client.advertisements.set(channel.id, channel);
```

Comment:
do we want to check if there are any existing channels for this id and make sure the client isn't overwriting them?